lib/iris/coords.py

```diff
@@ -1541,6 +1542,7 @@ def __init__(
         attributes=None,
         coord_system=None,
         climatological=False,
+        guess_coord=True,
```

Comment:
This is the thing that everyone was against in our first meeting, after which we reduced the proposal to a coordinate property that could be set, but not during instantiation.